docs/reference/model.rst

```diff
@@ -1,7 +1,82 @@
-.. automodule:: stravalib.model
-   :members:
-   :undoc-members:
-   :show-inheritance:
+====================================
+Model Module: Functions and Classes
+====================================
+.. currentmodule:: stravalib.model
 
+The ``Model`` does special things
 
-More here on this module
\ No newline at end of file
+Summary Functions
+-------------------------------
+.. autosummary::
+   :toctree: api/
+
+   lazy_property
+   check_valid_location
+   naive_datetime
+
+Summary Classes
```

Comment:
This approach allows us to better organize our classes and functions. Each class will then have a stub page with all of the details documented for it. 

There is some work to be done here to clean up docstrings and such. but i figure we can do one thing at a time. because that work will take some back and forth around descriptions and such.